### PR TITLE
fix(VSelect, VAutocomplete): modelValue respects [null]

### LINKED ISSUE #16442: [Bug Report][3.1.1] SelectItemKey doesn't support title for item with value = null
### Environment
**Vuetify Version:** 3.1.1
**Vue Version:** 3.2.45
**Browsers:** Chrome 108.0.0.0
**OS:** Mac OS 10.15.7

### Steps to reproduce
Create a v-select, add items as objects (with a title and a value) and use null as one of the values.
After that v-select renders empty instead of with the title for the null-value

[Recreation: play.vuetifyjs](https://play.vuetifyjs.com/#eNp9kk1P4zAQhv+K5Ut3pSYW+3HJlhVIIC7cEKe6B5NMisF2LNspQlX+O2M7oaEUbpl5Zt55Z5z1nl5aW+56oBVdBdBWiQD/uSFktSuEtekzBVpIM0Yp9qCgDlOCEKzoGlDnnI6EHpgSD4ncCrPtxRYOjIdK4liPUI3QzzsjLIIMCrDCCP3eygPHNMKdUH2ENU4/dLLJOJs5j0HeacVmu2LoaydtIB5CH7HUtnOB7ImDlgykdZ0mCzzSIhbXnfEBVxrdknOyjuJ7MhmsOL2CVvQqkMPCS8TZYmV6pciwnJrGLKdgivu7XNlKNcuVr0KrDN4nXJutkv6R01NKDRRX10dKKXdK6QacFmYS2vyLatOW+SlxRTzEj+j7J2LGTsBFcrqIfMXyOfGQdEnzLfERbPnkO4P/2T5NGAG+dkVSJubwxjHm9DEE6yvGemOft2XdaXaBjLneBKmhaDp98bv8Vf75yxrpwzxfgtfFg+tePDiciMvOxINsX78aMGLUPSvPsuyYipKfpBjq78AVDkwDDty3vo9qP3g/Yp/8x5kDNwPeUknz7I/OWPt0wvUmlgxLmqWxOO1BN2/d+EFC)

### Expected Behavior
Display the title for the null-value.

### Actual Behavior
Stays empty as no-value is selected

### Reproduction Link
[https://play.vuetifyjs.com/#...](https://play.vuetifyjs.com/#eNp9kk1P4zAQhv+K5Ut3pSYW+3HJlhVIIC7cEKe6B5NMisF2LNspQlX+O2M7oaEUbpl5Zt55Z5z1nl5aW+56oBVdBdBWiQD/uSFktSuEtekzBVpIM0Yp9qCgDlOCEKzoGlDnnI6EHpgSD4ncCrPtxRYOjIdK4liPUI3QzzsjLIIMCrDCCP3eygPHNMKdUH2ENU4/dLLJOJs5j0HeacVmu2LoaydtIB5CH7HUtnOB7ImDlgykdZ0mCzzSIhbXnfEBVxrdknOyjuJ7MhmsOL2CVvQqkMPCS8TZYmV6pciwnJrGLKdgivu7XNlKNcuVr0KrDN4nXJutkv6R01NKDRRX10dKKXdK6QacFmYS2vyLatOW+SlxRTzEj+j7J2LGTsBFcrqIfMXyOfGQdEnzLfERbPnkO4P/2T5NGAG+dkVSJubwxjHm9DEE6yvGemOft2XdaXaBjLneBKmhaDp98bv8Vf75yxrpwzxfgtfFg+tePDiciMvOxINsX78aMGLUPSvPsuyYipKfpBjq78AVDkwDDty3vo9qP3g/Yp/8x5kDNwPeUknz7I/OWPt0wvUmlgxLmqWxOO1BN2/d+EFC)

### PR DESCRIPTION
**Pull request by yuwu9145**
fixes #16442

## Description

- We use [Nullish coalescing operator (??)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) & `?? == null` extensively, it's beautiful and we should continue this approach by default, breaking it into checking `null` & `undefined` separately in different places would introduce inconsistent maintenance headache. 
- However, in terms of `item-value`, VCombobox supports null out-of-the-box ([link](https://play.vuetifyjs.com/#eNqdUrFu2zAQ/RWCi1vAEusG6KDKQQok6NKt6BR6oCXKZkNSBEmpDQT9e4+kZKuOkyGb7t7de/ee+Djgb8bkfcdxgUvPlZHM81uqESr7jBkTP2OhmNBTBfVxczsMyHHJK4/GsSTQQDN6muqzqlX7dt/+nVsIAVNbc7mlOG1TfMYk20fkB9OHjh34GaO+EHCeA1BOoFtuBjDzwksOE5qp0yr1FNoA9kx2AaxA/bxJZoNk4TAUyXtJFplA6SorjAffvguwUKa1Hg3I8gaNqLGtQisIcxWGq1Y7D5ama9EWPQbyAc0HFhTf84Z10qOz4TXA6cRCd1KicT0vTV2Kuc5+/UyTjZCLXv7MlEzASeFBH6RwR4qvMdU8u3+4YIq9a0zfuVVMz0S7r4GNakJQMjq9hW3I4kM4/SNMXIFW8dRVQEuS8oQk8RqnMOEvmPy3azU8yCFKTAD87gLFTuhByKGm+Oi9cQUhnTZPhxyeG7kDjNhOe6F4Vrfq7ia/yT9lTJojy7+QWji/hHPuVLa37R/HLQiD6YWGF83zazoTDPSbfPM58U69DB6yC8QvCAmo9NxmluuaW27fNHExe83IxcgLM0F6pHqEfKXQT+4i2srFWOPDDPU7XEaKsL8LMuMa/3cl6KZNvPsHgHxscw==)) and v2 supports null out-of-the-box as well ([link](https://codepen.io/yuchaosydney/pen/rNZbeRb)). So it also makes sense for v3 to support it consistently.
- To achieve this consistency, this PR introduce a new concept of "nullish symbol". When it is optionally applied to a certain senario like `item-value`, it internally transforms nullish values to a representative Symbol which makes nullish values internally  "legit" only within a certain scope. But eventually, v-model getter will revert "nullish symbol" back to its original value null. From user's perspective, it feels like null is "legit"

## Markup:

```vue
<template>
  <v-app>
    <v-main>
      <h1>{{ select }}</h1> 
      <v-select
        v-model="select"
        label="Language"
      	:items="languages"
        item-title="name"
    		item-value="code"
      />
<!-- <v-autocomplete
        v-model="select"
        label="Language"
      	:items="languages"
        item-title="name"
    		item-value="code"
      /> -->
    </v-main>
  </v-app>
</template>

<script setup>
import { ref } from 'vue'

const languages = [
  { "name":"Default Language", "code":null },
  { "code":"en-US", "file":"en-US.yaml", "name":"English" },
  { "code":"de-DE", "file":"de-DE.yaml", "name":"German" },
 ];
  
// const select = ref(null);
const select = ref('en-US');
</script>

```

### CODE REVIEW

**Comment by johnleider** [comment]
Can we write additional cypress tests that verify every other permutation of values that each supports?

**Comment by yuwu9145** [comment]
`getObjectValueByPath` does not use `Nullish coalescing operator` and is explicitly unhappy with `undefined`. 

https://github.com/vuetifyjs/vuetify/blob/cd06b48ce75b50c910267211ab3b7367053f8337/packages/vuetify/src/util/helpers.ts#L65

I thought this PR should resolve `undefined` as legit, but now it's convincible to exclude `undefined` and focus on `null` because undefined is not required and supporting it would introduce more breaking changes

File: packages/vuetify/src/components/VAutocomplete/VAutocomplete.tsx [hunk 1]
```diff
@@ -122,9 +122,13 @@ export const VAutocomplete = genericComponent<new <
       props,
       'modelValue',
       [],
-      v => transformIn(wrapInArray(v)),
       v => {
-        const transformed = transformOut(v)
+        const transformVal = revertSymbolIfNull(v)
+        return transformIn(transformVal == null ? [transformVal] : wrapInArray(transformVal))
```
Comment (johnleider) [inline packages/vuetify/src/components/VAutocomplete/VAutocomplete.tsx:127]: If we are no longer supporting undefined should this change?

File: packages/vuetify/src/components/VSelect/VSelect.tsx [hunk 2]
```diff
@@ -125,9 +125,13 @@ export const VSelect = genericComponent<new <
       props,
       'modelValue',
       [],
-      v => transformIn(wrapInArray(v)),
       v => {
-        const transformed = transformOut(v)
+        const transformVal = revertSymbolIfNull(v)
+        return transformIn(transformVal == null ? [transformVal] : wrapInArray(transformVal))
```
Comment (johnleider) [inline packages/vuetify/src/components/VSelect/VSelect.tsx:130]: If we are no longer supporting undefined should this change?

File: packages/vuetify/src/components/VAutocomplete/VAutocomplete.tsx (hunk 1 above)
Comment (yuwu9145) [reply packages/vuetify/src/components/VAutocomplete/VAutocomplete.tsx:127]: `transformVal == null` is necessary to allow null to be legit. Using `== null` is just consistent with all other places

**Comment by yuwu9145** [comment]
I personally feel this PR is overkill and over-engineered. At the beginning, I was trying to find a solution suites the whole framework. But supporting null is a case by case task so this PR is no longer the best candidate to tackle this issue.

Closing in favour of #16547

**Pull request closed without merging**